### PR TITLE
Add new component generators

### PR DESCRIPTION
**Pull request by job13er**
This #minor# change introduces new `component` generators

# CHANGELOG
 * **Removed** the `frost-component` generator
 * **Added** Some new dependencies that will be installed when this addon is `ember install`'d 
    * Addon: `ember-test-utils`
    * Bower packages: `chai-jquery` and `sinon-chai`
 * **Added** a `component` generator which uses our lint rules and our comment blocks
 * **Added** a `component-test` generator which uses `mocha` and `ember-test-utils`. It also sets up `sinon` automatically and has a failing test to make developers add a real test. 
 * **Added** a `component-addon` generator which does the re-export of a component in the `app` directory, basically the same as the default generator, minus the `;` at the end. 
 * **Added** a `PULL_REQUEST_TEMPLATE.md` file to help contributors remember to put version-bump comments and changelog messages in their PR descriptions.

### CODE REVIEW

File: blueprints/component-test/files/tests/__testType__/__path__/__test__.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/**
+ * <%= capitalizedTestType %> test for the <%= dasherizedModuleName %> component
+ */
+
+import {expect} from 'chai'
+import {describeComponent, it} from 'ember-mocha'
+<% if (testType === 'integration' ) { %>import hbs from 'htmlbars-inline-precompile'
+import {$hook, initialize} from 'ember-hook'
```
Comment (rox163) [inline blueprints/component-test/files/tests/__testType__/__path__/__test__.js]: could we rename the `initialize` function here, just to make it less ambiguous.
So `initialize as initializeHook` and update the call below ofcourse
We have been doing that for other addons as well during their LTS upgrades.

File: blueprints/component-addon/index.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+const stringUtil = require('ember-cli-string-utils')
+const validComponentName = require('ember-cli-valid-component-name')
+const path = require('path')
+
+module.exports = {
+  description: 'Generates a component. Name must contain a hyphen.',
+
+  /**
+   * Blueprint hook
+   * {@link https://ember-cli.com/extending/#filemaptokens}
```
Comment (sandersky) [inline blueprints/component-addon/index.js]: Looks like a good place to use `@see` http://usejsdoc.org/tags-see.html

**Review by sandersky — APPROVED** [review]

**Comment by sandersky** [comment]
👏 well done!

**Comment by sandersky** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/217/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

File: blueprints/component-test/files/tests/__testType__/__path__/__test__.js (hunk 1 above)
Comment (job13er) [reply blueprints/component-test/files/tests/__testType__/__path__/__test__.js]: Sure.

Comment (sglanzer-fire) [reply blueprints/component-test/files/tests/__testType__/__path__/__test__.js]: This is a good point, there are a fair number of `initialize` exports from modules

File: blueprints/component/files/__root__/__path__/__name__.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  // == Properties ============================================================
+
+  /**
+   * Properties for this component. Public properties are expected to be
+   * (potentially) passed in to the component. Private properties are *not*
+   * expected to be passed in/overwritten.
+   */
+  propTypes: {
+    // public
+    hook: PropTypes.string,
```
Comment (sglanzer-fire) [inline blueprints/component/files/__root__/__path__/__name__.js]: Can you make this required?

**Comment by job13er** [comment]
NOTE: the failing bit hound file is not one of mine. I believe it was already failing. We may want to address that, since otherwise we're merging failing builds :(

**Review by sglanzer-fire — CHANGES_REQUESTED** [review]
I'd just like to get the isRequired on the hook and I'm 👍

File: blueprints/component/files/__root__/__path__/__name__.js (hunk 3 above)
Comment (job13er) [reply blueprints/component/files/__root__/__path__/__name__.js]: Why? We have a default value for it. Making it required means that anytime anyone uses your component, they have to specify a hook. I know that for some components, ones I don't use more than once in a route, I don't normally set the hook, I just use the default.

**Review by sglanzer-fire — APPROVED** [review]

**Comment by sglanzer-fire** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/217/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

**Pull request merged**